src/core/featuremodel.cpp

```diff
@@ -57,6 +57,7 @@ void FeatureModel::setCurrentLayer( QgsVectorLayer *layer )
   mLayer = layer;
 
   connect( mLayer, &QgsVectorLayer::destroyed, this, &FeatureModel::removeLayer, Qt::UniqueConnection );
+  connect( mLayer, &QgsVectorLayer::featureAdded, this, &FeatureModel::featureAdded );
```

Comment:
Can you move this connection to line 333 and disconnect right after so it will not accidentally be triggered if some other code (future plugins...) add a feature?